### PR TITLE
Teleop Tests (includes, intake actuation, transversal climb, and shooter values)

### PR DESCRIPTION
**Pull request by WAaron-500**

### CODE REVIEW

File: src/main/java/frc/robot/commands/operatorcommands/TeleClimbTilt.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+public class TeleClimbTilt extends CommandBase {
+
+    // r 16
+
+    private final Climber climber;
+
+    public TeleClimbTilt(Climber climber) {
+        this.climber = climber;
+        climber.setBrakeMode(true);
```
Comment (hadley31) [inline src/main/java/frc/robot/commands/operatorcommands/TeleClimbTilt.java]: You should not be setting brakemode in the constructor. This should either be done in `RobotContainer`, or in the command's initialize method. But probably in robot container.

File: src/main/java/frc/robot/commands/operatorcommands/TeleClimbTilt.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    public void execute() {
+        climber.tiltRobot();
+    }
+
+    @Override
+    public void end(boolean interrupted) {
+    }
+
+    public boolean isFinished() {
+        return false;
```
Comment (hadley31) [inline src/main/java/frc/robot/commands/operatorcommands/TeleClimbTilt.java]:
```suggestion
        return true;
```

Comment (hadley31) [reply src/main/java/frc/robot/commands/operatorcommands/TeleClimbTilt.java]: We only need to execute once.

File: src/main/java/frc/robot/commands/operatorcommands/TeleClimbTilt.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    public TeleClimbTilt(Climber climber) {
+        this.climber = climber;
+        climber.setBrakeMode(true);
+    }
+
+    @Override
+    public void execute() {
+        climber.tiltRobot();
+    }
```
Comment (hadley31) [inline src/main/java/frc/robot/commands/operatorcommands/TeleClimbTilt.java]:
```suggestion
    private final Climber climber;
    private final boolean tilt;

    public TeleClimbTilt(Climber climber, boolean tilt) {
        this.climber = climber;
        this.tilt = tilt;
    }

    @Override
    public void execute() {
        climber.setTilt(tilt);
    }
```

File: src/main/java/frc/robot/subsystems/climber/Climber.java [hunk 4]
```diff
@@ -31,4 +31,11 @@ public void setRightPercent(double percent){
         rightClimberIO.setPercentPower(percent);
     }
 
+    public void tiltRobot() {
+        if(!leftClimberIO.getTilt()) {
+            leftClimberIO.tiltRobot(true);
+        } else if(leftClimberIO.getTilt()) {
+            leftClimberIO.tiltRobot(false);
+        }
+    }
```
Comment (hadley31) [inline src/main/java/frc/robot/subsystems/climber/Climber.java]:
```suggestion
    public void setTilt(boolean tilt) {
        if (tilt == this.tilt) {
            return;
        }
        
        this.tilt = tilt;
        leftClimberIO.setTilt(tilt);
        rightClimberIO.setTilt(tilt);
    }
```

File: src/main/java/frc/robot/subsystems/climber/ClimberIO.java [hunk 5]
```diff
@@ -10,4 +10,9 @@ public default void setPercentPower(double power){}
 
     public default void setTargetPoint(double degrees){}
 
+    public default void tiltRobot(boolean tilt) {}
```
Comment (hadley31) [inline src/main/java/frc/robot/subsystems/climber/ClimberIO.java]:
```suggestion
    public default void setTilt(boolean tilt) {}
```

File: src/main/java/frc/robot/subsystems/climber/ClimberIOFalcon.java [hunk 6]
```diff
@@ -36,4 +43,14 @@ public void setTargetPoint(double degrees){
         // Convert degrees to motor values
         climberMotor.set(ControlMode.Position, degrees);
     }
+
+    @Override
+    public void tiltRobot(boolean tilt) {
```
Comment (hadley31) [inline src/main/java/frc/robot/subsystems/climber/ClimberIOFalcon.java]:
```suggestion
    public void setTilt(boolean tilt) {
```

**Review by hadley31 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/RobotContainer.java [hunk 7]
```diff
@@ -1,85 +1,122 @@
...
 
-    // Define button / command bindings here
-    controls.getClimbUp().whileActiveOnce(climberUpCommand);
-    controls.getClimbDown().whileActiveOnce(climberDownCommand);
-  }
+        // Define button / command bindings here
+        controls.getClimbUp().whileActiveOnce(climberUpCommand);
+        controls.getClimbDown().whileActiveOnce(climberDownCommand);
+        controls.getClimbButton().whileActiveOnce(climmberTiltCommand);
+        controls.getIntakeRetractButton().whileActiveOnce(intakeToggleCommand);
```
Comment (hadley31) [inline src/main/java/frc/robot/RobotContainer.java]:
```suggestion
        controls.getIntakeRetractButton().whenActive(intakeToggleCommand);
```

File: src/main/java/frc/robot/commands/operatorcommands/TeleClimbDown.java [hunk 8]
```diff
@@ -17,13 +17,11 @@ public TeleClimbDown(Climber climber) {
     @Override
     public void execute() {
         climber.setLeftPercent(-0.7);
```
Comment (hadley31) [inline src/main/java/frc/robot/commands/operatorcommands/TeleClimbDown.java]: This should be renamed to indicate both are being set, and not just left. Even something like `setPercent()`

File: src/main/java/frc/robot/commands/operatorcommands/TeleClimbTilt.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    // r 16
+
+    private final Climber climber;
+
+    public TeleClimbTilt(Climber climber) {
+        this.climber = climber;
+    }
+
+    @Override
+    public void initialize() {
```
Comment (hadley31) [inline src/main/java/frc/robot/commands/operatorcommands/TeleClimbTilt.java]:
```suggestion
    public void execute() {
```

File: src/main/java/frc/robot/commands/operatorcommands/TeleClimbTilt.java [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        climber.tiltRobot();
+        System.out.println("command innit");
+    }
+
+    @Override
+    public void end(boolean interrupted) {
+    }
+
+    public boolean isFinished() {
+        return false;
```
Comment (hadley31) [inline src/main/java/frc/robot/commands/operatorcommands/TeleClimbTilt.java]:
```suggestion
        return true;
```

File: src/main/java/frc/robot/commands/operatorcommands/TeleClimbTilt.java (hunk 9 above)
Comment (hadley31) [reply src/main/java/frc/robot/commands/operatorcommands/TeleClimbTilt.java]: We should be doing all command activities in the execute method. Initialize should do just that... Initialize anything that needs to happen before the command is executed. Then `return true` in `isFinished()`

File: src/main/java/frc/robot/commands/operatorcommands/TeleFlyVarDown.java [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    @Override
+    public void execute() {
+        varFlyWheel.retractFly();
+    }
+    
+    @Override
+    public void end(boolean interrupted) {
+        varFlyWheel.stop();
```
Comment (hadley31) [inline src/main/java/frc/robot/commands/operatorcommands/TeleFlyVarDown.java]: Why are we stopping the flywheel after retracting?

File: src/main/java/frc/robot/commands/operatorcommands/TeleFlyVarDown.java [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    @Override
+    public boolean isFinished() {
+        return false;
```
Comment (hadley31) [inline src/main/java/frc/robot/commands/operatorcommands/TeleFlyVarDown.java]:
```suggestion
        return true;
```

Comment (hadley31) [reply src/main/java/frc/robot/commands/operatorcommands/TeleFlyVarDown.java]: One and done command?

File: src/main/java/frc/robot/commands/operatorcommands/TeleFlyVarUp.java [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    @Override
+    public void execute() {
+        varFlyWheel.extendFly();
+    }
+    
+    @Override
+    public void end(boolean interrupted) {
+        varFlyWheel.stop();
```
Comment (hadley31) [inline src/main/java/frc/robot/commands/operatorcommands/TeleFlyVarUp.java]: Again why are we stopping wheel? Should this be a parallel command group?

File: src/main/java/frc/robot/commands/operatorcommands/TeleFlyVarUp.java [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    @Override
+    public boolean isFinished() {
+        return false;
```
Comment (hadley31) [inline src/main/java/frc/robot/commands/operatorcommands/TeleFlyVarUp.java]:
```suggestion
        return true;
```

Comment (hadley31) [reply src/main/java/frc/robot/commands/operatorcommands/TeleFlyVarUp.java]: One and done?

File: src/main/java/frc/robot/commands/operatorcommands/TeleIntakeToggle.java [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+    // r 16
+
+    private final Intake intake;
+
+    public TeleIntakeToggle(Intake intake) {
+        this.intake = intake;
+    }
+
+    @Override
+    public void initialize() {
```
Comment (hadley31) [inline src/main/java/frc/robot/commands/operatorcommands/TeleIntakeToggle.java]:
```suggestion
    public void execute() {
```

File: src/main/java/frc/robot/commands/operatorcommands/TeleIntakeToggle.java [hunk 16]
```diff
@@ -0,0 +1,30 @@
+package frc.robot.commands.operatorcommands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.climber.Climber;
+import frc.robot.subsystems.intake.Intake;
+
+public class TeleIntakeToggle extends CommandBase {
+
+    // r 16
+
```
Comment (hadley31) [inline src/main/java/frc/robot/commands/operatorcommands/TeleIntakeToggle.java]:
```suggestion
```

**Review by hadley31 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/operatorcommands/TeleFlyVarDown.java (hunk 11 above)
Comment (WAaron-500) [reply src/main/java/frc/robot/commands/operatorcommands/TeleFlyVarDown.java]: oops

File: src/main/java/frc/robot/commands/operatorcommands/TeleFlyVarUp.java (hunk 13 above)
Comment (WAaron-500) [reply src/main/java/frc/robot/commands/operatorcommands/TeleFlyVarUp.java]: fixed

**Pull request closed without merging**